datafusion/core/tests/physical_optimizer/filter_pushdown/mod.rs

```diff
@@ -267,7 +267,7 @@ async fn test_dynamic_filter_pushdown_through_hash_join_with_topk() {
         format_plan_for_test(&plan),
         @r"
     - SortExec: TopK(fetch=2), expr=[e@4 ASC], preserve_partitioning=[false], filter=[e@4 IS NULL OR e@4 < bb]
-    -   HashJoinExec: mode=Partitioned, join_type=Inner, on=[(a@0, d@0)], filter=[d@0 >= aa AND d@0 <= ab]
+    -   HashJoinExec: mode=Partitioned, join_type=Inner, on=[(a@0, d@0)]
```

Comment:
To other reviewers, the filters are still pushed down to the DataSourceExec (it just isn't reflected on the HashJoinExec itself0